docs: summarize behavior of routes files (#11720)
sveltejs/kit: documentation/docs/20-core-concepts/10-routing.md

Each route directory contains one or more _route files_, which can be identified by their `+` prefix.

> Here are a few simple rules to help you remember how SvelteKit's routing works:

Reviewer: ```suggestion
We'll introduce these files in a moment in more detail, but here are a few simple rules to help you remember how SvelteKit's routing works:
```

> * All files can run on the **server**
> * Files run on the **client**, except `+server` files
> * `+layout` and `+error` files affect routes in subdirectories

Reviewer: ```suggestion
* All files can run on the server
* All files run on the client except `+server` files
* `+layout` and `+error` files affect the rendering of routes in subdirectories
```

> * `+layout` and `+error` files affect the rendering of routes in subdirectories

Reviewer: ```suggestion
* `+layout` and `+error` files apply to subdirectories as well as the directory they live in
```

Each route directory contains one or more _route files_, which can be identified by their `+` prefix.

Summary of the routing files:

> | File name         | Role                                                                | Propagates to sub-dirs | SSR | CSR |

Reviewer: SSR and CSR don't make sense for the API endpoints. You can invoke endpoints while doing server side rendering or client-side rendering. I'm guessing you're referring to the endpoint code itself running on either the client or server
```suggestion
| File name         | Role                                                                | Propagates to sub-dirs | Runs on server | Runs on client |
```

> | File name         | Role                                                                | Propagates to sub-dirs | Runs on server | Runs on client |

Reviewer: ```suggestion
| File name         | Role                                                                | Propagates to subdirectories | Runs on server | Runs on client |
```

|-------------------|---------------------------------------------------------------------|:----------------------:|:---:|:---:|

> | +page.svelte      | Create an HTML page                                                 |           ❌            |  ✅  |  ✅  |
> | +error.svelte     | Create an HTML page to render HTTP errors                           |           ✅            |  ✅  |  ✅  |
> | +layout.svelte    | Create an HTML layout                                               |           ✅            |  ✅  |  ✅  |
> | +page.js          | Provide data for the **page** via the `load()` function             |           ❌            |  ✅  |  ✅  |

Reviewer: These should have an "s" at the end. E.g. you would say "+page.svelte creates an HTML page"
```suggestion
| +page.svelte      | Creates an HTML page                                                |           ❌            |  ✅  |  ✅  |
| +error.svelte     | Creates an HTML page to render HTTP errors                          |           ✅            |  ✅  |  ✅  |
| +layout.svelte    | Creates an HTML layout                                              |           ✅            |  ✅  |  ✅  |
| +page.js          | Provides data for the **page** via the `load()` function            |           ❌            |  ✅  |  ✅  |
```

| +page.server.js   | API endpoint for this **page**, automatically called at page load   |           ❌            |  ✅  |  ❌  |

> | +server.js        | API endpoint for this **page**, to be manually called via `fetch()` |           ❌            |  ✅  |  ❌  |

Reviewer: This isn't just for a page. It can be called from an external service, etc.
```suggestion
| +server.js        | API endpoint                                                     |           ❌            |  ✅  |  ❌  |
```

> |-------------------|-------------------------------------------------------|:----------------------:|:---:|:---:|
> | +page.svelte      | HTML page                                             |            ❌           |  ✅  |  ✅  |
> | +error.svelte     | HTML page to render HTTP errors                       |            ✅           |  ✅  |  ✅  |
> | +layout.svelte    | Layout used. Needs at least a <slot /> tag            |            ✅           |  ✅  |  ✅  |
> | +page.js          | Send public data to HTML page                         |            ❌           |  ✅  |  ✅  |
> | +page.server.js   | Send private data to HTML page                        |            ❌           |  ✅  |  ❌  |
> | +server.js        | API endpoint to be called from HTML pages             |            ❌           |  ✅  |  ❌  |
> | +layout.js        | Send public data to HTML page covered by this layout  |            ✅           |  ✅  |  ✅  |
> | +layout.server.js | Send private data to HTML page covered by this layout |            ✅           |  ✅  |  ❌  |

Reviewer: I haven't formed an opinion about this idea yet, but if we did do it then I think many of the descriptions should be different
```suggestion
|-------------------|-------------------------------------------------------|:----------------------:|:---:|:---:|
| +page.svelte      | Generates HTML page                                   |            ❌           |  ✅  |  ✅  |
| +error.svelte     | Generates HTML page when error encountered             |            ✅           |  ✅  |  ✅  |
| +layout.svelte    | Generates layout HTML                                       |            ✅           |  ✅  |  ✅  |
| +page.js          | Provide data for page via `load`                      |            ❌           |  ✅  |  ✅  |
| +page.server.js   | API endpoint for page. Automatically called             |            ❌           |  ✅  |  ❌  |
| +server.js        | API endpoint to be called manually                    |            ❌           |  ✅  |  ❌  |
| +layout.js        | Provide data for layout via `load`                    |            ✅           |  ✅  |  ✅  |
| +layout.server.js | API endpoint for layout. Automatically called          |            ✅           |  ✅  |  ❌  |
```

Author: Not sure about "Generate", but your suggestions are good, I'll make an update and suggest another clearer wording 👍

Reviewer: > Also, why do you think it's necessary to specify "Automatically called", while it's already ticked in the SSR column?

By "automatically called", I mean you don't have to write a `load` function and call `fetch` yourself. I don't see what that has to do with SSR

> | +layout.js        | Provide data for the **layout** via the `load()` function           |           ✅            |  ✅  |  ✅  |

Reviewer: ```suggestion
| +layout.js        | Provides data for the **layout** via the `load()` function           |           ✅            |  ✅  |  ✅  |
```

> | +page.js          | Provides data for the **page** via the `load()` function            |           ❌            |       ✅        |       ✅        |
> | +page.server.js   | API endpoint for this **page**, automatically called at page load   |           ❌            |       ✅        |       ❌        |
> | +server.js        | API endpoint                                                        |           ❌            |       ✅        |       ❌        |
> | +layout.js        | Provides data for the **layout** via the `load()` function          |           ✅            |       ✅        |       ✅        |
> | +layout.server.js | API endpoint for this **layout**, automatically called at page load |           ✅            |       ✅        |       ❌        |

Reviewer: I don't really see a reason to bold those words
```suggestion
| +page.js          | Provides data for the page via the `load()` function                |           ❌            |       ✅        |       ✅        |
| +page.server.js   | API endpoint for this page, automatically called at page load       |           ❌            |       ✅        |       ❌        |
| +server.js        | API endpoint                                                        |           ❌            |       ✅        |       ❌        |
| +layout.js        | Provides data for the layout via the `load()` function              |           ✅            |       ✅        |       ✅        |
| +layout.server.js | API endpoint for this layout, automatically called at page load     |           ✅            |       ✅        |       ❌        |
```
